src/common/util/include/openvino/util/file_util.hpp

```diff
@@ -172,26 +161,30 @@ bool directory_exists(const std::string& path);
 bool directory_exists(const std::wstring& path);
 #endif
 
+inline ov::util::Path cut_android_path(const ov::util::Path& file_name) {
+    const auto& file_name_result = file_name.native();
+    const auto pos = file_name_result.find('!');
+    if (pos != std::decay_t<decltype(file_name_result)>::npos) {
+        return ov::util::Path(file_name_result.substr(0, pos));
+    }
+
+    return file_name;
+}
+
 /**
  * @brief      Returns file size for file
  * @param[in]  path  The file name
  * @return     file size
  */
-inline int64_t file_size(const char* path) {
-#if defined(OPENVINO_ENABLE_UNICODE_PATH_SUPPORT) && defined(_WIN32)
-    std::wstring widefilename = ov::util::string_to_wstring(path);
-    const wchar_t* file_name = widefilename.c_str();
-#elif defined(__ANDROID__) || defined(ANDROID)
-    std::string file_name = path;
-    std::string::size_type pos = file_name.find('!');
-    if (pos != std::string::npos) {
-        file_name = file_name.substr(0, pos);
-    }
+inline int64_t file_size(const ov::util::Path& path) {
```

Comment:
```suggestion
inline int64_t file_size(const ov::util::Path& path) {
    #if defined(__ANDROID__) || defined(ANDROID)
    // apply cuting if std::filesytem not support it well.
    #endfi
    return std::filesystem::file_size(path);
}
```
This function should probably be wrapper to `std` implementation if works.

Reply:
fix